lxd/instances_post.go

```diff
@@ -88,12 +88,19 @@ func createFromImage(s *state.State, r *http.Request, p api.Project, profiles []
 	}
 
 	run := func(op *operations.Operation) error {
+		profileDevices := make(map[string]map[string]string)
+		for _, p := range profiles {
+			for k, v := range p.Devices {
+				profileDevices[k] = v
+			}
+		}
+
 		args := db.InstanceArgs{
 			Project:     p.Name,
 			Config:      req.Config,
 			Type:        dbType,
 			Description: req.Description,
-			Devices:     deviceConfig.NewDevices(req.Devices),
+			Devices:     deviceConfig.NewDevicesWithInitialValues(req.Devices, profileDevices),
```

Comment:
How would you feel about altering `NewDevicesWithInitialValues` to `ApplyDeviceInitialValues` and then changing the usage to this:

```go
devices := deviceConfig.NewDevices(req.Devices)

args := db.InstanceArgs{
...
    Devices:     deviceConfig.ApplyDeviceInitialValues(devices, profiles)
...
}
```

This avoids duplicating the loop convering the profiles to a map, and avoids combining discrete unrelated steps into the same function `NewDevicesWithInitialValues`.